nighthawk/test/options_test.cc

```diff
@@ -0,0 +1,171 @@
+#include "gtest/gtest.h"
+
+#include "test/test_common/utility.h"
+
+#include "nighthawk/source/client/options_impl.h"
+
+using namespace std::chrono_literals;
+
+namespace Nighthawk {
+namespace Client {
+
+class OptionsImplTest : public testing::Test {
+public:
+  OptionsImplTest()
+      : client_name_("nighthawk_client"), good_test_uri_("http://127.0.0.1/"),
+        no_arg_match_("Couldn't find match for argument") {}
+
+  std::unique_ptr<OptionsImpl> createOptionsImpl(const std::string& args) {
+    std::vector<std::string> words = Envoy::TestUtility::split(args, ' ');
+    std::vector<const char*> argv;
+    for (const std::string& s : words) {
+      argv.push_back(s.c_str());
+    }
+    return std::make_unique<OptionsImpl>(argv.size(), argv.data());
+  }
+
+  std::string client_name_;
+  std::string good_test_uri_;
+  std::string no_arg_match_;
+};
+
+TEST_F(OptionsImplTest, BogusInput) {
+  // TODO(oschaaf): enable this test after we have host name validation.
+  // EXPECT_THROW_WITH_REGEX(createOptionsImpl(fmt::format("{} --foo", client_name_)),
+  //                        MalformedArgvException, no_arg_match_);
+}
+
+TEST_F(OptionsImplTest, All) {
+  std::unique_ptr<OptionsImpl> options =
+      createOptionsImpl(fmt::format("{} --rps 4 --connections 5 --duration 6 --timeout 7 --h2 "
+                                    "--concurrency 8  --verbosity error {}",
+                                    client_name_, good_test_uri_));
+
+  EXPECT_EQ(4, options->requests_per_second());
+  EXPECT_EQ(5, options->connections());
+  EXPECT_EQ(6s, options->duration());
+  EXPECT_EQ(7s, options->timeout());
+  EXPECT_EQ(true, options->h2());
+  EXPECT_EQ("8", options->concurrency());
+  EXPECT_EQ("error", options->verbosity());
+  EXPECT_EQ(good_test_uri_, options->uri());
+
+  // Check that our conversion to CommandLineOptionsPtr makes sense.
+  CommandLineOptionsPtr cmd = options->toCommandLineOptions();
+  EXPECT_EQ(cmd->requests_per_second(), options->requests_per_second());
+  EXPECT_EQ(cmd->connections(), options->connections());
+  EXPECT_EQ(cmd->duration().seconds(), options->duration().count());
+  EXPECT_EQ(cmd->timeout().seconds(), options->timeout().count());
+  EXPECT_EQ(cmd->h2(), options->h2());
+  EXPECT_EQ(cmd->concurrency(), options->concurrency());
+  EXPECT_EQ(cmd->verbosity(), options->verbosity());
+  EXPECT_EQ(cmd->uri(), options->uri());
+}
+
+// TODO(oschaaf): A couple of the tests below are pretty repetitive. They perform
+// the same sanity checking with just the argument name changing.
+// Figure out how to deduplicate.
```

Comment:
Are these candidates for `TEST_P` parameterized tests?

Reply:
Yes; with `TEST_P` this looks much better, thanks. I planned to circle back to this myself but forgot to.